### PR TITLE
persqueue: add user-agent counters

### PR DESCRIPTION
**Pull request by qyryq**

### CODE REVIEW

File: ydb/services/persqueue_v1/actors/helpers.cpp [hunk 1]
```diff
@@ -27,4 +27,27 @@ bool HasMessages(const Topic::StreamReadMessage::ReadResponse& data) {
...
+        case '*':
+        case '?':
+        case '"':
+        case '\'':
+        case '`':
+        case '\\':
+            continue;
+        default:
+            clean.push_back(c);
+            if (clean.size() == 200) {
```
Comment (qyryq) [inline ydb/services/persqueue_v1/actors/helpers.cpp:45]: move the magic number to a properly named variable

File: ydb/services/persqueue_v1/actors/helpers.cpp [hunk 2]
```diff
@@ -27,4 +27,27 @@ bool HasMessages(const Topic::StreamReadMessage::ReadResponse& data) {
     return false;
 }
 
+
+TString CleanupCounterValueString(const TString& value) {
+    TString clean;
+    for (auto c : value) {
+        switch (c) {
+        case '|':
```
Comment (qyryq) [inline ydb/services/persqueue_v1/actors/helpers.cpp:35]: explain this set of characters

File: ydb/services/persqueue_v1/actors/write_session_actor.cpp [hunk 3]
```diff
@@ -488,6 +492,17 @@ void TWriteSessionActor<UseMigrationProtocol>::InitAfterDiscovery(const TActorCo
...
 
+template<bool UseMigrationProtocol>
+void TWriteSessionActor<UseMigrationProtocol>::SetupBytesWrittenByUserAgentCounter() {
+    static constexpr auto protocol = UseMigrationProtocol ? "pqv1" : "topic";
+    BytesWrittenByUserAgent = GetServiceCounters(Counters, "pqproxy|userAgents")
+        ->GetSubgroup("host", "")
+        ->GetSubgroup("protocol", protocol)
+        ->GetSubgroup("topic", FullConverter->GetFederationPath())
+        ->GetSubgroup("user-agent", CleanupCounterValueString(UserAgent))
+        ->GetExpiringNamedCounter("sensor", "BytesWrittenByUserAgent", true);
```
Comment (qyryq) [inline ydb/services/persqueue_v1/actors/write_session_actor.cpp:503]: bytes_written_by_user_agent?

File: ydb/services/persqueue_v1/actors/write_session_actor.cpp [hunk 4]
```diff
@@ -488,6 +492,17 @@ void TWriteSessionActor<UseMigrationProtocol>::InitAfterDiscovery(const TActorCo
...
 }
 
+template<bool UseMigrationProtocol>
+void TWriteSessionActor<UseMigrationProtocol>::SetupBytesWrittenByUserAgentCounter() {
+    static constexpr auto protocol = UseMigrationProtocol ? "pqv1" : "topic";
+    BytesWrittenByUserAgent = GetServiceCounters(Counters, "pqproxy|userAgents")
+        ->GetSubgroup("host", "")
+        ->GetSubgroup("protocol", protocol)
+        ->GetSubgroup("topic", FullConverter->GetFederationPath())
+        ->GetSubgroup("user-agent", CleanupCounterValueString(UserAgent))
```
Comment (qyryq) [inline ydb/services/persqueue_v1/actors/write_session_actor.cpp:502]: user_agent?

**Review by alexnick88 — DISMISSED** [review]

**Review by alexnick88 — APPROVED** [review]

**Pull request merged**